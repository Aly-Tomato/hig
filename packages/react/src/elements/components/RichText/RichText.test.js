import renderer from "react-test-renderer";
import React from "react";
import { RichText } from "../../../hig-react";

describe("RichText", () => {
  it("default", () => {
    const tree = renderer
      .create(
        <RichText>
          The RichText component lets me include any React children, including:
          <ul>
            <li>
              <strong>Any old HTML tag</strong>
            </li>
            <li>React components</li>
          </ul>
        </RichText>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with prop size small", () => {
    const tree = renderer
      .create(
        <RichText size="small">
          The RichText component lets me include any React children, including:
          <ul>
            <li>
              <strong>Any old HTML tag</strong>
            </li>
            <li>React components</li>
          </ul>
        </RichText>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with prop size large", () => {
    const tree = renderer
      .create(
        <RichText size="large">
          The RichText component lets me include any React children, including:
          <ul>
            <li>
              <strong>Any old HTML tag</strong>
            </li>
            <li>React components</li>
          </ul>
        </RichText>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
