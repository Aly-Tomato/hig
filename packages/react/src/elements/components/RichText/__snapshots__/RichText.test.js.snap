// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichText default 1`] = `
<div
  className="hig__rich-text"
>
  The RichText component lets me include any React children, including:
  <ul>
    <li>
      <strong>
        Any old HTML tag
      </strong>
    </li>
    <li>
      React components
    </li>
  </ul>
</div>
`;

exports[`RichText renders correctly with prop size large 1`] = `
<div
  className="hig__rich-text hig__rich-text--large"
>
  The RichText component lets me include any React children, including:
  <ul>
    <li>
      <strong>
        Any old HTML tag
      </strong>
    </li>
    <li>
      React components
    </li>
  </ul>
</div>
`;

exports[`RichText renders correctly with prop size small 1`] = `
<div
  className="hig__rich-text hig__rich-text--small"
>
  The RichText component lets me include any React children, including:
  <ul>
    <li>
      <strong>
        Any old HTML tag
      </strong>
    </li>
    <li>
      React components
    </li>
  </ul>
</div>
`;
